### PR TITLE
Added sceKernelGetModuleList

### PR DESCRIPTION
**Pull request by CoreyHendrey**
Tried to implement sceKernelGetModuleList

Called in CUSA09316 (https://github.com/shadps4-emu/shadps4-game-compatibility/issues/136)

### CODE REVIEW

**Comment by CoreyHendrey** [comment]
Hmm, I didn't touch the failing file on macOs

**Comment by viniciuslrangel** [comment]
> Hmm, I didn't touch the failing file on macOs

Prob function was used somewhere which had <algorithm> included. Move implementation of this function https://github.com/shadps4-emu/shadPS4/blob/c74d6535468462823c7b0de6cfe5766c9be268be/src/core/module.h#L168-L175 to the .cpp file

File: src/core/linker.h [hunk 1]
```diff
@@ -79,9 +78,7 @@ class Linker {
         return m_modules.at(index).get();
     }
 
-    int GetNumberModules() const {
-        return m_modules.size();
```
Comment (viniciuslrangel) [inline src/core/linker.h]: This is ok to keep

**Comment by Hermiten** [comment]
Is it finish ? @ItsStolas @viniciuslrangel 
Thanks you :)

File: src/core/libraries/kernel/libkernel.cpp [hunk 2]
```diff
@@ -398,6 +399,23 @@ int PS4_SYSV_ABI posix_getpagesize() {
     return 4096;
 }
 
+int PS4_SYSV_ABI sceKernelGetModuleList(Core::Module** pArray, size_t numArray,
```
Comment (raphaelthegreat) [inline src/core/libraries/kernel/libkernel.cpp:402]: Core::Module is our structure, not sure it matches what guest expects

**Comment by polybiusproxy** [comment]
Closing due to inactivity.

**Pull request closed without merging**